Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_tbag.nut

```diff
@@ -0,0 +1,397 @@
+global function GamemodeTbag_Init
+global function CreateBattery
+
+struct {
+	table<entity, float> players // needed for detecting if tbag grace period exceeded 3 seconds
+	table<entity, array<entity> > trigger // needed for detecting individual batteries
+	table<entity, int> tbag // needed for detecting tbag counter
+	int floatingDuration // how long it takes for the battery to disappear
+	int teabagCounter // how many times do you need to teabag
+} file
+
+
+void function GamemodeTbag_Init()
+{
+	SetShouldUseRoundWinningKillReplay( true )
+	Riff_ForceTitanAvailability( eTitanAvailability.Never )
+	ClassicMP_ForceDisableEpilogue( true )
+
+	AddCallback_OnClientConnected( TbagInitPlayer )
+	AddCallback_OnPlayerKilled( TbagOnPlayerKilled )
+	AddCallback_OnClientDisconnected( TbagCleanupClient )
+	AddCallback_GameStateEnter( eGameState.WinnerDetermined, OnWinnerDetermined )
+
+	SetTimeoutWinnerDecisionFunc( CheckScoreForDraw )
+
+	file.floatingDuration = GetCurrentPlaylistVarInt( "floatingduration", 30 )
+	file.teabagCounter = GetCurrentPlaylistVarInt( "teabagcounter", 4 )
+
+	AddClientCommandCallback("Tbag_down", Crouching );
+	AddClientCommandCallback("Tbag_downtoggle", ToggleCrouching ); // toggle crouch requires x2 the input for whatever reason
+}
+
+void function TbagInitPlayer( entity player )
+{
+	file.players[player] <- Time() // start beginning countdown
+	file.tbag[player] <- 0 // newslot this so the tbag counter starts at 0
+}
+
+void function TbagCleanupClient( entity player )
+{
+	// remove all saved tables of this player when they disconnect.
+	if (player in file.players)
+		delete file.players[player]
+
+	if (player in file.trigger)
+		delete file.trigger[player]
+
+	if (player in file.tbag)
+		delete file.tbag[player]
+}
+
+void function TbagOnPlayerKilled( entity victim, entity attacker, var damageInfo )
+{
+	if ( !victim.IsPlayer() || GetGameState() != eGameState.Playing || attacker == victim)
+		return
+
+	if ( attacker.IsPlayer() )
+    {
+		CreateBattery(victim)
+		SetRoundWinningKillReplayAttacker(attacker)
+    }
+}
+
+void function OnWinnerDetermined()
+{
+	SetRespawnsEnabled( false )
+	SetKillcamsEnabled( false )
+}
+
+int function CheckScoreForDraw()
+{
+	if (GameRules_GetTeamScore(TEAM_IMC) > GameRules_GetTeamScore(TEAM_MILITIA))
+		return TEAM_IMC
+	else if (GameRules_GetTeamScore(TEAM_MILITIA) > GameRules_GetTeamScore(TEAM_IMC))
+		return TEAM_MILITIA
+
+	return TEAM_UNASSIGNED
+}
+
+// ========================================================================================================
+//
+// BATTERY STUFF
+//
+// ========================================================================================================
+
+void function CreateBattery(entity player)
+{
+	entity batteryPack = CreateEntity( "prop_dynamic" )
+	batteryPack.SetValueForModelKey( RODEO_BATTERY_MODEL ) // ideally I would want the Helmet collectible from the campaign :/

```

Comment:
any particular reason why you can't use the helmet collectible? i was going to suggest using it myself

Reply:
Couldn't find the model path lol.

Reply:
should be $"models/humans/heroes/mlt_hero_jack_helmet_static.mdl", which is defined in mp/_pickups.gnut as `const HELMET_COLLECTIBLE_MODEL = $"models/humans/heroes/mlt_hero_jack_helmet_static.mdl"` (pickups.gnut is an sp-only script for some fucking reason though despite being in the mp folder????)

Reply:
![image](https://user-images.githubusercontent.com/22678145/184615985-a73a3999-ce8c-4e53-9a01-19c6b62ed9ef.png)

The model doesn't seem to be in MP even though I already did `PrecacheModel` sadly.